### ISSUE TITLE
RFE for Fl_Menu_Button to allow dropdown to appear left OR right aligned

### ISSUE THREAD

**Issue by erco77** [fltk/fltk#362]
This would be relevant to improving the look of the Fl_Input_Choice widget, which uses Fl_Menu_Button as a simple arrow-down box.

As Albrecht wrote in a previous issue, what's being requested (in ascii art):
Current behavior:

```
+------------+---+
| one        | v |
+------------+---+
             +---------+
             | one     |
             | two     |
             | three   |
             +---------+
```

Requested behavior:

```
+------------+---+
| one        | v |
+------------+---+
       +---------+
       | one     |
       | two     |
       | three   |
       +---------+

```
..where `| v |` is the Fl_Menu_Button, and `| one    |` is a separate Fl_Input widget.

So, regardless of the actual width of the Fl_Menu_Button, the dropdown menu's right edge would align with the button's right edge, instead of left aligned to the button's left edge.

**Comment by spitzak** [comment]
The menu is supposed to go from the left edge of the text field to the right edge of the downarrow button, not sure when this got changed.

**Comment by erco77** [comment]
The Fl_Menu_Button widget is fine with respect to itself (the behavior is as you describe it).

But in the context of Fl_Input_Choice, it's an Fl_Group with an Fl_Input and small Fl_Menu_Button next to it (w/h=20x20).

So the idea is to be able to get the Fl_Menu_Button pulldown menu's position to be sized to appear under the entire Fl_Input_Choice::Fl_Group,. Perhaps there can be a way to set the x position and width of the pulldown menu so it appears directly under the entire Fl_Input_Choice area.

**Comment by erco77** [comment]
Attaching a suggested patch [input_choice-v1.patch.txt](https://github.com/fltk/fltk/files/7865364/input_choice-v1.patch.txt) that causes the pulldown menu to appear below the entire width of the Fl_Input_Choice widget as shown in the screenshot below. If the menu has contents wider than the widget, it extends off to the right.

![screenshot-patch-v1](https://user-images.githubusercontent.com/6484779/149395697-9cc64a65-a05e-49d6-ab86-5faf5d511ccd.png)

The patch as shown here modifies existing behavior; it does not offer an API to change the existing behavior to this. Which leads me to this:

Question: Should this be the new default behavior? It looks pretty nice and is what most people expect. Or should we keep the old behavior and make this new behavior an option that has to be turned on? The change doesn't break user code, just changes where the menu appears.

**Comment by spitzak** [comment]
I believe this behavior is always wanted.

**Comment by erco77** [comment]
A +1 from me as well.. it's what I would have done way back when I implemented it (before I became a dev), and had I known how to do it. Awaiting Albrecht's input.

**Comment by MatthiasWM** [comment]
+1. Love it!

**Comment by Albrecht-S** [comment]
:+1: +1 from me as well as **the** new behavior and no option to change it. I consider this a bug fix. We might want to document it but I wouldn't even do this, just change it.

The only thing we need to make sure is that it works well if the menu appears near the screen border (current behavior is to adjust the menu that it fits in the screen). I tested it and it looks good. I also noticed the comment "(This is the same handle() code in Fl_Menu_Button and Fl_Choice)" so it should be fine (?).

There's one typo though in the posted patch which breaks compilation: `s/InputMenu_Button/InputMenuButton/`. I wonder how this could happen?
